test/valid-data/string-literals-intrinsic/main.ts

```diff
@@ -4,11 +4,13 @@ type ResultUpper = Uppercase<Result>;
 type ResultLower = Lowercase<ResultUpper>;
 type ResultCapitalize = Capitalize<Result>;
 type ResultUncapitalize = Uncapitalize<ResultCapitalize>;
+type ResultNoInfer = NoInfer<Result>;
```

Comment:
I'm pretty sure this isn't how `NoInfer` is intended to be used and is a flaky test since all other `intrinsic` types are string-dependent and `NoInfer` might not be.

Please create a separate test with a code like:

```ts
// (I wrote this out of my head)

function test<T>(value: NoInfer<T>): T { return value } 
// create a function with type constraints as well (test<T extends something>())

export const MyObject = {
  str: test('asd'),
  // more samples with objects, type references, and so on...
}

```

Reply:
but thanks for working in this <3